### PR TITLE
feat(auth): l10n changes for Mozilla Account branding (auth-server)

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We want l10n strings that are changing from Firefox account(s) to Mozilla account(s) to be localized before launch

## This pull request

* Create new copies of ftl strings that contained Firefox account/accounts and replace brand term with Mozilla account/accounts
* This ticket DOES NOT implement the new strings, that will be handled in a follow-up ticket (8321)

## Issue that this pull request solves

Closes: #FXA-7992

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Brand terms (`-product-mozilla-account` and `-product-mozilla-accounts`) are already added to fxa-shared for use by fxa-auth-server (among other packages).

Skipped updates to recovery key emails as they will be updated with account-agnostic content in JIRA 7669

### CODE REVIEW

**Comment by vpomerleau** [comment]
Hi @bcolsson! I hesitated between a few different ways of adding the new (and currently unused) strings to the FTL files but ultimately went with just adding the new strings directly under the old ones (while making sure to copy existing comments). 

The other options I considered might make cleanup easier, but seemed harder to review in PR and would increase line diff (alternative options: https://mozilla-hub.atlassian.net/browse/FXA-7992?focusedCommentId=730207)

If you think an alternate presentation would be better for l10n, please let me know! Thanks

File: packages/fxa-auth-server/lib/l10n/server.ftl [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ## Non-email strings
 
 session-verify-send-push-title = Logging in to { -product-firefox-accounts }?
+session-verify-send-push-title-2 = Logging in to { -product-mozilla-accounts }?
```
Comment (bcolsson) [inline packages/fxa-auth-server/lib/l10n/server.ftl:4]:
```suggestion
session-verify-send-push-title-2 = Logging in to your { -product-mozilla-account }?
```

According to the marketing guielines around the rebrand, `account` is almost always singular and plural should only be used when referring to changes to all Mozilla accounts. So in this case we should go with singular.

File: packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/en.ftl [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+passwordChangeRequired-suspicious-activity-2 = We detected suspicious behavior on your { -product-mozilla-account }. To prevent unauthorized access to your { -product-mozilla-account }, we’ve disconnected all devices on your account and are requiring you to change your password as a precaution.
 passwordChangeRequired-sign-in = Sign back into any device or service where you use your { -product-firefox-account } and follow the steps that will be presented to you.
+passwordChangeRequired-sign-in-2 = Sign back into any device or service where you use your { -product-mozilla-account } and follow the steps that will be presented to you.
 passwordChangeRequired-different-password = <b>Important:</b> Pick a different password than what you were previously using and make sure that it is different from your email account.
 passwordChangeRequired-signoff = Best,
 passwordChangeRequired-signoff-name = The { -product-firefox-accounts } team
+passwordChangeRequired-signoff-name-2 = The { -product-mozilla-accounts } team
```
Comment (bcolsson) [inline packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/en.ftl:10]: We might want to check with the content team if we're okay using `The Mozilla accounts team` or if we should avoid using `Mozilla accounts` here.

File: packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl [hunk 3]
```diff
@@ -1,7 +1,10 @@
 fraudulentAccountDeletion-subject = Your { -product-firefox-account } was deleted
+fraudulentAccountDeletion-subject-2 = Your { -product-mozilla-account } was deleted
 fraudulentAccountDeletion-title = Your account was deleted
 fraudulentAccountDeletion-content = Recently, a { -product-firefox-account } was created and a subscription was charged using this email address. As we do with all new accounts, we asked that you confirm your account by first validating this email address.
+fraudulentAccountDeletion-content-moz = Recently, a { -product-mozilla-account } was created and a subscription was charged using this email address. As we do with all new accounts, we asked that you confirm your account by first validating this email address.
```
Comment (bcolsson) [inline packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl:5]:
```suggestion
fraudulentAccountDeletion-content-2 = Recently, a { -product-mozilla-account } was created and a subscription was charged using this email address. As we do with all new accounts, we asked that you confirm your account by first validating this email address.
```

Nit: only this and one other string are updated using `-moz`. Suggestion is for consistency.

File: packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl [hunk 4]
```diff
@@ -1,7 +1,10 @@
...
 fraudulentAccountDeletion-content-2 = At present, we see that the account was never confirmed. Since this step was not completed, we are not sure if this was an authorized subscription. As a result, the { -product-firefox-account } registered to this email address was deleted and your subscription was canceled with all charges reimbursed.
+fraudulentAccountDeletion-content-2-moz = At present, we see that the account was never confirmed. Since this step was not completed, we are not sure if this was an authorized subscription. As a result, the { -product-mozilla-account } registered to this email address was deleted and your subscription was canceled with all charges reimbursed.
```
Comment (bcolsson) [inline packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl:7]:
```suggestion
fraudulentAccountDeletion-content-3 = At present, we see that the account was never confirmed. Since this step was not completed, we are not sure if this was an authorized subscription. As a result, the { -product-mozilla-account } registered to this email address was deleted and your subscription was canceled with all charges reimbursed.
```

Nit: only this and one other string are updated using -moz. Suggestion is for consistency.

File: packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl (hunk 3 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl:5]: This one was tricky, because `fraudulentAccountDeletion-content`and `fraudulentAccountDeletion-content-2` are both currently in use, and not different versions of the same text. Perhaps the new version could be `fraudulentAccountDeletion-content-part1-v2` and `fraudulentAccountDeletion-content-part2-v2`?

File: packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/en.ftl (hunk 2 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/en.ftl]: @bcolsson I got signoff from Vesta on removing the signoff block entirely from this email template 👍

File: packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl (hunk 4 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl]: See comment above  - we need to leave `fraudulentAccountDeletion-content` and `fraudulentAccountDeletion-content-2` untouched, so I've replaced the new string IDs with  `fraudulentAccountDeletion-content-part1-v2` and `fraudulentAccountDeletion-content-part2-v2`

**Review by bcolsson — APPROVED** [review]

File: packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl (hunk 4 above)
Comment (bcolsson) [reply packages/fxa-auth-server/lib/senders/emails/templates/fraudulentAccountDeletion/en.ftl]: Your change makes sense and IDs are easier to understand 👍

File: packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/en.ftl [hunk 5]
```diff
@@ -4,4 +4,5 @@ cadReminderFirst-action-plaintext = { cadReminderFirst-action }:
 # In the title of the email, "It takes two to sync", "two" refers to syncing two devices
 cadReminderFirst-title-1 = It takes two to sync
 cadReminderFirst-description-1 = Take your tabs across all your devices. Get your bookmarks, passwords, and other data everywhere you use { -brand-firefox }. It’s like having magic in your { -brand-firefox } account!
+cadReminderFirst-description-part1-v2 = Take your tabs across all your devices. Get your bookmarks, passwords, and other data everywhere you use { -brand-firefox }.
```
Comment (dschom) [inline packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/en.ftl:7]: Why `-v2` instead of `-2` postfix? Is it because the l10n id already ends with a number?

File: packages/fxa-auth-server/lib/senders/emails/templates/newDeviceLogin/en.ftl [hunk 6]
```diff
@@ -4,6 +4,8 @@ newDeviceLogin-subject = New sign-in to { $clientName }
 # Variables:
 # $clientName (String) - A client the user hasn't signed into before (e.g. Firefox, Sync)
 newDeviceLogin-title-2 = Your { -product-firefox-account } was used to sign in
+# $clientName (String) - A client the user hasn't signed into before (e.g. Firefox, Sync)
```
Comment (dschom) [inline packages/fxa-auth-server/lib/senders/emails/templates/newDeviceLogin/en.ftl:7]: I think comment is just extra copy/paste. There's no $clientName variable in the the `newDeviceLogin-title-3` string.

File: packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/index.mjml [hunk 7]
```diff
@@ -23,15 +23,5 @@
     <mj-text css-class="text-body align-left">
       <span data-l10n-id="passwordChangeRequired-different-password"><b>Important:</b> Pick a different password than what you were previously using and make sure that it is different from your email account.</span>
     </mj-text>
-
```
Comment (dschom) [inline packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/index.mjml:26]: Nice, clean up.

Also, just double checking that this should have been removed. Seems unnecessary. I think I saw this in a thread somewhere.

File: packages/fxa-auth-server/lib/senders/emails/templates/verifyShortCode/en.ftl [hunk 8]
```diff
@@ -2,7 +2,10 @@
 #  $code (Number) - e.g. 123456
 verifyShortCode-subject-3 = Confirm your account
 verifyShortCode-title-2 = Open the internet with { -brand-firefox }
+verifyShortCode-title-3 = Open the internet with { -brand-mozilla }
```
Comment (dschom) [inline packages/fxa-auth-server/lib/senders/emails/templates/verifyShortCode/en.ftl:5]: Was this a string you double checked with UX? I can see this being debatable. I think it makes sense, but just double checking here.

File: packages/fxa-auth-server/package.json [hunk 9]
```diff
@@ -47,7 +47,7 @@
     "subscription-reminders": "CONFIG_FILES='config/secrets.json' node -r esbuild-register ./scripts/subscription-reminders.ts",
     "audit-orphaned-stripe-accounts": "CONFIG_FILES='config/secrets.json' node -r esbuild-register ./scripts/audit-orphaned-customers.ts",
     "remove-unverified-accounts": "CONFIG_FILES='config/secrets.json' node -r esbuild-register ./scripts/remove-unverified-accounts.ts",
-    "storybook": "NODE_OPTIONS=--openssl-legacy-provider storybook dev -p 6010 --no-version-updates -s ./",
+    "storybook": "NODE_OPTIONS=--openssl-legacy-provider storybook dev -p 6010 --no-version-updates ./",
```
Comment (dschom) [inline packages/fxa-auth-server/package.json:50]: Good catch on a deprecated flag. Should we configure more specific static dirs? Definitely not a show stopper, just something for another time.

File: packages/fxa-auth-server/lib/senders/emails/templates/verificationReminderFinal/en.ftl [hunk 10]
```diff
@@ -1,4 +1,5 @@
 verificationReminderFinal-subject = Final reminder to confirm your account
 verificationReminderFinal-description = A couple of weeks ago you created a { -product-firefox-account }, but never confirmed it. For your security, we will delete the account if not verified in the next 24 hours.
+verificationReminderFinal-description-2 = A couple of weeks ago you created a { -product-mozilla-account }, but never confirmed it. For your security, we will delete the account if not verified in the next 24 hours.
```
Comment (dschom) [inline packages/fxa-auth-server/lib/senders/emails/templates/verificationReminderFinal/en.ftl:3]: I think we should deal with potential confusion here, but I can definitely seeing this being confusing in the couple of weeks following our rollout.

File: packages/fxa-auth-server/lib/senders/emails/templates/postVerify/index.stories.ts [hunk 11]
```diff
@@ -16,6 +16,7 @@ const createStory = storyWithProps(
     link: 'http://localhost:3030/connect_another_device',
     desktopLink: 'https://firefox.com',
     onDesktopOrTabletDevice: true,
+    productName: 'Firefox',
```
Comment (dschom) [inline packages/fxa-auth-server/lib/senders/emails/templates/postVerify/index.stories.ts:19]: Just curious with everything being .ftl changes, why was this needed?

**Review by dschom — APPROVED** [review]
Looks good to me. A couple questions / sanity checks, but looks great overall. Thanks for all the updates!

File: packages/fxa-auth-server/lib/senders/emails/templates/postVerify/index.stories.ts (hunk 11 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/postVerify/index.stories.ts:19]: This was a fix so this story would render (missing prop was causing it to error)

File: packages/fxa-auth-server/lib/senders/emails/templates/verificationReminderFinal/en.ftl (hunk 10 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/verificationReminderFinal/en.ftl:3]: Good point! Should we consider adding logic so that the Mozilla-branded email includes the string with `Firefox accounts` if the account was created before the rollout date?

File: packages/fxa-auth-server/package.json (hunk 9 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/package.json:50]: We have staticDirs also configured in `.storybook/main.js`, so launching the storybook was resulting in an error (needed to configure static dirs in only one spot). So I think we're covered here and no follow-up required :)

File: packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/index.mjml (hunk 7 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/passwordChangeRequired/index.mjml:26]: Correct! Confirmed with UX :)

File: packages/fxa-auth-server/lib/senders/emails/templates/newDeviceLogin/en.ftl (hunk 6 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/newDeviceLogin/en.ftl:7]: Good catch! I'll remove from the original string too.

File: packages/fxa-auth-server/grunttasks/ftl.js [hunk 12]
```diff
@@ -5,6 +5,17 @@
 'use strict';
 
 module.exports = function (grunt) {
+  // make a copy of the local branding terms available for local development
```
Comment (vpomerleau) [inline packages/fxa-auth-server/grunttasks/ftl.js:8]: Pulled in this grunt task that was recently added to fxa-settings to make sure the latest branding file is available for local dev

File: packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/en.ftl [hunk 13]
```diff
@@ -4,4 +4,4 @@ cadReminderFirst-action-plaintext = { cadReminderFirst-action }:
 # In the title of the email, "It takes two to sync", "two" refers to syncing two devices
 cadReminderFirst-title-1 = It takes two to sync
 cadReminderFirst-description-1 = Take your tabs across all your devices. Get your bookmarks, passwords, and other data everywhere you use { -brand-firefox }. It’s like having magic in your { -brand-firefox } account!
-cadReminderFirst-description-2 = It only takes a sec to sync.
```
Comment (vpomerleau) [inline packages/fxa-auth-server/lib/senders/emails/templates/cadReminderFirst/en.ftl:7]: Verified this change with UX - this sentence was not quite accurate, and removing it simplifies the copy and removes the need for adding `part1` to the new version of previous ftl string. 
Kept the v2 though to make sure this will be flagged as a new string for localization.

File: packages/fxa-auth-server/lib/senders/emails/layouts/fxa/en.ftl [hunk 14]
```diff
@@ -4,7 +4,10 @@
 ## version. The strings are usually identical but sometimes they differ slightly.
 
 fxa-header-firefox-logo = <img data-l10n-name="fxa-logo" alt="{ -brand-firefox } logo">
+fxa-header-mozilla-logo = <img data-l10n-name="mozilla-logo" alt="{ -brand-mozilla } logo">
 fxa-header-sync-devices-image = <img data-l10n-name="sync-devices-image" alt="Sync devices">
 body-devices-image = <img data-l10n-name="devices-image" alt="Devices">
 fxa-privacy-url = { -brand-mozilla } Privacy Policy
+moz-accounts-privacy-url = { -product-mozilla-accounts(capitalization:"uppercase") } Privacy Policy
```
Comment (vpomerleau) [inline packages/fxa-auth-server/lib/senders/emails/layouts/fxa/en.ftl:11]: @dschom @LZoog - Example of parametrization in ftl strings to pick between two different options of the same term (set up in `branding.ftl`)

Based on "parametrized terms" section of: 
https://projectfluent.org/fluent/guide/terms.html

File: packages/fxa-auth-server/lib/senders/emails/templates/verifyShortCode/en.ftl (hunk 8 above)
Comment (vpomerleau) [reply packages/fxa-auth-server/lib/senders/emails/templates/verifyShortCode/en.ftl:5]: Just posted a question in the slack channel, I had not confirmed this one.

File: packages/fxa-shared/l10n/branding.ftl [hunk 15]
```diff
@@ -25,7 +25,12 @@
 -product-mozilla-account = Mozilla account
 
 # "accounts" can and should be localized, "Mozilla" must be treated as a brand. Plural "Mozilla accounts" is used when referring to something affecting all Mozilla accounts, not just the individual's account.
--product-mozilla-accounts = Mozilla accounts
+# uppercase variant is primaly used for Title case, and legal contexts
+-product-mozilla-accounts =
```
Comment (vpomerleau) [inline packages/fxa-shared/l10n/branding.ftl:29]: I've added a capitalization option for the rare case where we need to uppercase "Mozilla Accounts", with "Mozilla accounts" as default.

@bcolsson I'm realizing now that this might need a new id though since the string has already been extracted... Not sure if that is best, or if I should add a new uppercase term instead in addition to the existing one.

**Review by vpomerleau — COMMENTED** [review]
@dschom Thanks very much for the review! I've made a few follow up changes (incl. adding copy grunt task for auth-server and adding a proposal for uppercasing "Mozilla Accounts", mainly for legal terms).

File: packages/fxa-shared/l10n/branding.ftl (hunk 15 above)
Comment (bcolsson) [reply packages/fxa-shared/l10n/branding.ftl:29]: I gave this some thought, I think we'll want to parameterize the existing term. Since this change is mostly a capitalization change which is locale independent it doesn't warrant a new ID. This also creates potentially redundant terms for many locales that have different scripts or styles. 

When we do add the parameter to the term, I'll send a message to locales about the update to the term just to cover our bases here.

File: packages/fxa-shared/l10n/branding.ftl [hunk 16]
```diff
@@ -25,7 +25,12 @@
 -product-mozilla-account = Mozilla account
 
 # "accounts" can and should be localized, "Mozilla" must be treated as a brand. Plural "Mozilla accounts" is used when referring to something affecting all Mozilla accounts, not just the individual's account.
--product-mozilla-accounts = Mozilla accounts
+# We almost always want "accounts" lowercased, however, there is at least one special case where we consider capitalized "Accounts" as headline casing of a legal document or reference to a legal document.
```
Comment (LZoog) [inline packages/fxa-shared/l10n/branding.ftl:28]: @vpomerleau @bcolsson Given the recent decision about capitalizing "A" also for the email "sender", which will be done in a follow up, maybe we should update the comment in case that may change a translation for the uppercase version? Or should we update the comment then, or leave as-is?

Comment (bcolsson) [reply packages/fxa-shared/l10n/branding.ftl:28]:
```suggestion
# "accounts" should be lowercase in almost all cases. Uppercase is reserved for special use cases where headline case is necessary, for example legal document names and references.
```
We could adjust the comment, here's an example. I think we'd want to rely on comments explaining why capitalized "accounts" is used for the strings we are using the `uppercase` parameter.
